### PR TITLE
Clean up LoggingTensor semantic

### PR DESCRIPTION
**Pull request by albanD**
Clarify how LoggingTensor works with autograd.
The updated comment should cover the semantic changes.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72623
* #72622
* __->__ #72620

Differential Revision: [D34214956](https://our.internmc.facebook.com/intern/diff/D34214956)

### CODE REVIEW

File: torch/testing/_internal/logging_tensor.py [hunk 1]
```diff
@@ -22,6 +22,13 @@ def no_dispatch() -> Iterator[None]:
 # 3. Enter dispatcher, wind your way through Autograd
 # 4. Hit Python dispatch key, call __torch_dispatch__
 
+# This Tensor can work with autograd in two ways:
```
Comment (albanD) [inline torch/testing/_internal/logging_tensor.py:25]: Update this to say that allowing both is for testing only. And you should not mix the two in a single test or you're going to get very weird behavior!

Comment (zou3519) [reply torch/testing/_internal/logging_tensor.py:25]: Question: are you actually planning on using LoggingTensor in both ways for testing?

**Review by zou3519 — APPROVED** [review]
LGTM. A note that mixing the two will save future test writers from a lot of confusion :)

File: torch/testing/_internal/logging_tensor.py (hunk 1 above)
Comment (albanD) [reply torch/testing/_internal/logging_tensor.py:25]: Yes. The current test do use the logging Tensor having gradients.
And the new ones added above in this stack check for the wrapped Tensor requirering gradients

**Pull request closed without merging**